### PR TITLE
feat(core): Get page of blocked logins

### PR DESCRIPTION
**Pull request by bodnara**
* created method to get paginated blocked logins for all layers
* it is possible to filter by login and multiple namespaces
* created object for blocked logins query and column sorting

### CODE REVIEW

File: perun-base/src/main/java/cz/metacentrum/perun/core/api/BlockedLoginsPageQuery.java [hunk 1]
```diff
@@ -0,0 +1,124 @@
+package cz.metacentrum.perun.core.api;
+
+import java.util.List;
+import java.util.Objects;
+
+/**
+ * Class representing a query requesting a specific page of blocked logins.
+ *
+ * @author Metodej Klang
```
Comment (Johaney-s) [inline perun-base/src/main/java/cz/metacentrum/perun/core/api/BlockedLoginsPageQuery.java:9]:
```suggestion
 * @author Adam Bodnar
```
:)

File: perun-core/src/main/java/cz/metacentrum/perun/core/impl/UsersManagerImpl.java [hunk 2]
```diff
@@ -1249,6 +1247,46 @@ public void unblockLogin(PerunSession sess, String login, String namespace) thro
 		}
 	}
 
+	@Override
+	public Paginated<Pair<String, String>> getBlockedLoginsPage(PerunSession sess, BlockedLoginsPageQuery query) {
+		try {
+			MapSqlParameterSource namedParams = new MapSqlParameterSource();
+			namedParams.addValue("login", query.getSearchString());
+			namedParams.addValue("namespaces", query.getNamespaces());
+			namedParams.addValue("order", query.getSortColumn().getSqlOrderBy(query));
```
Comment (Johaney-s) [inline perun-core/src/main/java/cz/metacentrum/perun/core/impl/UsersManagerImpl.java:1256]: not needed anymore

File: perun-core/src/test/java/cz/metacentrum/perun/core/entry/UsersManagerEntryIntegrationTest.java [hunk 3]
```diff
@@ -871,6 +840,84 @@ public void blockAlreadyUsedLogin() throws Exception {
...
+
+		Paginated<Pair<String, String>> blockedLogins = usersManager.getBlockedLoginsPage(sess, query);
+		assertNotNull(blockedLogins);
+		assertTrue(blockedLogins.getData().size() > 1);
+		assertTrue(blockedLogins.getData().containsAll(List.of(new Pair<String, String>(login, namespace), new Pair<String, String>(login2, null))));
+	}
+
+	@Test
+	public void getBlockedLoginPage_searchString() throws Exception {
+		System.out.println(CLASS_NAME + "getBlockedLoginsPage_all");
```
Comment (Johaney-s) [inline perun-core/src/test/java/cz/metacentrum/perun/core/entry/UsersManagerEntryIntegrationTest.java:864]:
```suggestion
		System.out.println(CLASS_NAME + "getBlockedLoginPage_searchString");
```

File: perun-rpc/src/main/java/cz/metacentrum/perun/rpc/methods/UsersManagerMethod.java [hunk 4]
```diff
@@ -1200,6 +1201,20 @@ public Void call(ApiCaller ac, Deserializer parms) throws PerunException {
 		}
 	},
 
+	/*#
+	 * Get a page of blocked logins
+	 *
+	 * @param query BlockedLoginsPageQuery
```
Comment (sarkapalkovicova) [inline perun-rpc/src/main/java/cz/metacentrum/perun/rpc/methods/UsersManagerMethod.java:1207]: Missing return tag

**Review by sarkapalkovicova — COMMENTED** [review]
Looks good, but it will be helpful if the method also returns the IDs of logins since we use them for unblocking. Otherwise, the unblockLogins method will need to be edited again.

**Review by sarkapalkovicova — APPROVED** [review]

**Review by mattjoke — APPROVED** [review]
Looks good 👍

**Pull request merged**